script/bootstrap

```diff
@@ -56,17 +60,19 @@ install_packages_apt()
         sudo apt-get --no-install-recommends install -y gcc-arm-embedded
     fi
 
+    echo 'Installing pretty tools useful for code contributions...'
+
     # add clang-format for pretty
     sudo apt-get --no-install-recommends install -y clang-format-6.0
 
     # add yapf for pretty
-    python3 -m pip install yapf==0.29.0 || echo 'Failed to install python code formatter yapf. Install it manually if you need.'
+    python3 -m pip install yapf==0.29.0 || echo 'WARNING: could not install yapf, which is useful if you plan to contribute python code to the OpenThread project.'
```

Comment:
```suggestion
    python3 -m pip install yapf==0.29.0 || echo 'WARNING: could not install yapf, which is useful if you plan to contribute to the OpenThread project.'
```
[Coding Conventions and Style](https://github.com/openthread/openthread/blob/master/CONTRIBUTING.md#coding-conventions-and-style) requires developers to use `script/make-pretty` and `script/make-pretty check` to format or check code. Whether or not Python code is modified, the script will fail if `yapf` is not  installed. So, I think `yapf` is required for contributing any code.

Reply:
`script/make-pretty` has options to select which checks you want to run. The options make it possible to avoid using `yapf` if you are not formatting python code.
```
#     script/make-pretty clang
#     script/make-pretty markdown
#     script/make-pretty python
#     script/make-pretty shell
#
#     script/make-pretty check clang
#     script/make-pretty check markdown
#     script/make-pretty check python
#     script/make-pretty check shell
```

Reply:
Yes it's possible to not check Python code when it's not changed, but how to check only one type of code is not taught in CONTRIBUTING.md. 

A contributor who want to contribute some shell code would conclude that `yapf` is not required after reading this error messages. Then, if the contributor simply follows the [Coding Conventions and Style](https://github.com/openthread/openthread/blob/master/CONTRIBUTING.md#coding-conventions-and-style), he or she will end up in checking errors. 

So, I think we should make sure contributors who follow guides strictly do not fail unexpectedly.

Reply:
I think it is more helpful to be provide more info in the warning message about the use of tool (specially in cae when it is not installed).

`"WARNING: could not install yapf, which is useful if you plan to contribute python code to the OpenThread project"`

seems a lot more useful/informative  to me compared to:

`'WARNING: could not install yapf, which is useful if you plan to contribute to the OpenThread project.'`

I am sure OT contributors are skilled and capable enough to be able to figure out how to use `make-pretty` script (it actually has good documentation in the script itself).